cmd/main.go

```diff
@@ -139,3 +161,101 @@ func main() {
 
 	os.Exit(0)
 }
+
+func runMultishareReconciler(driverConfig *driver.GCFSDriverConfig) {
+	config, err := buildConfig(*kubeconfig)
+	if err != nil {
+		klog.Error(err.Error())
+		os.Exit(1)
+	}
+
+	kubeClient, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		klog.Error(err.Error())
+		os.Exit(1)
+	}
+	fsClient, err := clientset.NewForConfig(config)
+	if err != nil {
+		klog.Error(err.Error())
+		os.Exit(1)
+	}
+
+	factory := fsInformers.NewSharedInformerFactory(fsClient, resyncPeriod)
+	coreFactory := informers.NewSharedInformerFactory(kubeClient, resyncPeriod)
+	sharescheme.AddToScheme(scheme.Scheme)
+
+	recon := reconciler.NewMultishareReconciler(
+		fsClient,
+		driverConfig,
+		factory.Multishare().V1alpha1().ShareInfos(),
+		factory.Multishare().V1alpha1().InstanceInfos(),
+		coreFactory.Storage().V1().StorageClasses().Lister(),
+	)
+
+	if err := ensureCustomResourceDefinitionsExist(fsClient); err != nil {
+		klog.Errorf("Exiting due to failure to ensure CRDs exist during startup: %+v", err)
+		os.Exit(1)
+	}
+
+	run := func(context.Context) {
+		// run...
+		stopCh := make(chan struct{})
+		factory.Start(stopCh)
+		coreFactory.Start(stopCh)
+		go recon.Run(stopCh)
+
+		// ...until SIGINT
+		c := make(chan os.Signal, 1)
+		signal.Notify(c, os.Interrupt)
+		<-c
+		close(stopCh)
+	}
+
+	go run(context.TODO())
+}
+
+// Checks that the ShareInfo v1alpha1 CRDs exist.
+func ensureCustomResourceDefinitionsExist(client *clientset.Clientset) error {
+	condition := func() (bool, error) {
+		var err error
+
+		// scoping to an empty namespace makes `List` work across all namespaces
+		_, err = client.MultishareV1alpha1().ShareInfos().List(context.TODO(), metav1.ListOptions{})
+		if err != nil {
+			klog.Errorf("Failed to list v1alpha1 shareinfos with error=%+v", err)
+			return false, nil
+		}
+
+		return true, nil
+	}
+
+	// The maximum retry duration = initial duration * retry factor ^ # steps. Rearranging, this gives
+	// # steps = log(maximum retry / initial duration) / log(retry factor).
+	const retryFactor = 1.5
```

Comment:
please create some well defined constants for these

Reply:
done in most recent code https://github.com/leiyiz/gcp-filestore-csi-driver/blob/stateful_csi/pkg/csi_driver/gcfs_driver.go